shell/src/peer_manager.rs

```diff
@@ -336,6 +336,13 @@ impl Actor for PeerManager {
         });
     }
 
+    fn post_start(&mut self, ctx: &Context<Self::Msg>) {
+        if !self.initial_peers.is_empty() {
+            info!(ctx.system.log(), "Connecting to defined peers");
+            self.initial_peers.drain().for_each(|address| ctx.myself().tell(ConnectToPeer { address: address.clone() }, ctx.myself().into()));
```

Comment:
@adonagy 
hmm, drain will clear self.initial_peers, so later they wont be used on `discover_peers(`,
this is not good, we need to find betrer way to initialize connections on startup

Reply:
@bkontur  Well, we could just iterate through `self.initial_peers` instead of drain. That would ensure the Set is not cleared. The idea behind drain was, that when a peer (from `self.initial_peers`) disconnects, or won't even connect there is no point keeping it in initial peers. Ideally when this happens, we would want to look for an another peer. As the name suggests, they are the initial peers (or "starting" peers)